dpnp/dpnp_iface_indexing.py

```diff
@@ -614,6 +788,94 @@ def indices(
     return res
 
 
+def mask_indices(
+    n,
+    mask_func,
+    k=0,
+    device=None,
+    usm_type="device",
+    sycl_queue=None,
+):
+    """
+    Return the indices to access (n, n) arrays, given a masking function.
+
+    Assume `mask_func` is a function that, for a square array a of size
+    ``(n, n)`` with a possible offset argument `k`, when called as
+    ``mask_func(a, k=k)`` returns a new array with zeros in certain locations
+    (functions like `triu` or `tril` do precisely this). Then this function
+    returns the indices where the non-zero values would be located.
+
+    Parameters
+    ----------
+    n : int
+        The returned indices will be valid to access arrays of shape (n, n).
+    mask_func : callable
+        A function whose call signature is similar to that of `triu`, `tril`.
```

Comment:
I'd propose to make a link on a page with documentation of functions:
```suggestion
        A function whose call signature is similar to that of :obj:`dpnp.triu`, :obj:`dpnp.tril`.
```